### ISSUE TITLE
Error when using distlib==0.2.0

### ISSUE THREAD

**Issue by StoicLoofah** [brettcannon/caniusepython3#76]
So I'm not sure where the bug is, but I ran into this bug when running caniusepython3:

```
kevin@myserver:~$ /usr/local/bin/caniusepython3 -r requirements.txt
Finding and checking dependencies ...
Traceback (most recent call last):
  File "/usr/local/bin/caniusepython3", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python2.7/dist-packages/caniusepython3/__main__.py", line 159, in main
    check(projects_from_cli(args))
  File "/usr/local/lib/python2.7/dist-packages/caniusepython3/__main__.py", line 145, in check
    blockers = dependencies.blocking_dependencies(projects, pypi.all_py3_projects())
  File "/usr/local/lib/python2.7/dist-packages/caniusepython3/dependencies.py", line 80, in blocking_dependencies
    dist = distlib.locators.locate(project)
  File "/usr/local/lib/python2.7/dist-packages/distlib/locators.py", line 361, in locate
    result.download_urls = versions.get('urls', {}).get(version, set())
AttributeError: 'dict' object has no attribute 'download_urls'
```

I fixed it by rolling back to distlib==0.1.9, though that gave me these errors

```
Finding and checking dependencies ...
[WARNING] recaptcha not found  
[WARNING] django-recaptcha-field not found
[ERROR] Failed to get external data for https://www.red-dove.com/pypi/projects/E/email-reply-parser/project.json: HTTP Error 404: Not Found
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/distlib/util.py", line 744, in _get_external_data
    resp = urlopen(url)
  File "/usr/lib/python2.7/urllib2.py", line 126, in urlopen
    return _opener.open(url, data, timeout)
  File "/usr/lib/python2.7/urllib2.py", line 406, in open
    response = meth(req, response)
  File "/usr/lib/python2.7/urllib2.py", line 519, in http_response
    'http', request, response, code, msg, hdrs)
  File "/usr/lib/python2.7/urllib2.py", line 444, in error
    return self._call_chain(*args)
  File "/usr/lib/python2.7/urllib2.py", line 378, in _call_chain
    result = func(*args)
  File "/usr/lib/python2.7/urllib2.py", line 527, in http_error_default
    raise HTTPError(req.get_full_url(), code, msg, hdrs, fp)
HTTPError: HTTP Error 404: Not Found

You need 17 projects to transition to Python 3.
Of those 17 projects, 15 have no direct dependencies blocking their transition:

  django-celery-transactions   
  django-concurrent-server
  django-discover-jenkins
  django-include-strip-tag
  django-ipware
  django-storages
  docx
  hellosign-python-sdk
  oauth2
  pdfminer
  pykss
  reportlab
  lepl (which is blocking rfc6266)
  m2crypto (which is blocking tender_multipass)
  xlsxcessive
```

Not sure if this is a problem with distlib, my setup, or maybe freezing the required version of distlib in the requirements. Happy to report more as necessary. If it helps, I'm running python 2.7.3 in ubuntu 12.04

**Comment by brettcannon** [comment]
Tests all pass fine under distlib 0.2.0, so it isn't obvious what the problem is. Can you upload your requirements.txt file so I can more accurately reproduce your situation?

**Comment by StoicLoofah** [comment]
Sure. I'm not sure what the state of it is or if it's relevant, but when I tried this requirement file on caniusepython3.com, it also hung

```
django==1.7.4
# django-haystack 2.3.0 and up support django 1.7
django-haystack==2.3.1
pysolr==2.0.15
python-memcached==1.47
reportlab==2.5  # apt-get install python-dev libfreetype6-dev
PyPDF2==1.19  # used for watermarking
pdfminer==20131113  # used for text extraction because PyPDF2 is really bad ZAN-6127
pyenchant==1.6.5  # apt-get install enchant
lxml==3.2.1  # apt-get install libxml2-dev libxslt-dev
openpack==1.0  # For XlsXcessive
XlsXcessive==0.1.6  # external+unverified, use provided wheel or --allow-external XlsXcessive --allow-unverified XlsXcessive
httpagentparser==1.5.1
django-include-strip-tag==0.1.0
psycopg2==2.5.1  # apt-get install postgresql-9.1
pillow==2.5.3
tender-multipass==0.2.1  # apt-get install python-m2crypto, then link global M2Crypto to virtualenv
httplib2==0.8  # for oauth2
oauth2==1.5.211
xlrd==0.9.0
docx==0.2.0
pycountry==0.18
six==1.4.1
# below are required for django-celery
anyjson==0.3.3  # celery requirement
amqp==1.4.6  # celery requirement
billiard==3.3.0.19  # celery requirement
celery==3.1.17  # django-celery requirement
kombu==3.0.24  # django-celery requirement
python-dateutil==2.1  # django-celery requirement
PyTZ==2013b  # django-celery requirement
django-celery==3.1.15
recaptcha==1.0rc1
django-recaptcha-field==1.0b2
boto==2.13.3
django-storages==1.1.8
futures==2.1.3
django-pipeline==1.3.23
django-flashpolicies==1.5
sorl-thumbnail==11.12.1b
celery-haystack==0.8
django-appconf==0.6  # celery-haystack requirement
django-celery-transactions==0.1.3  # celery-haystack requirement
### Testing
### Jenkins
logilab-common==0.62.1  # django-jenkins requirement, may be deprecated
coverage==3.7.1  # django-discover-jenkins component
pylint==1.4.0  # django-discover-jenkins component
astroid==1.3.2  # pylint requirement
django-discover-jenkins==0.1.2
pep8==0.6.1
selenium==2.44.0
PyVirtualDisplay==0.1.2
EasyProcess==0.1.6  # PyVirtualDisplay requirement
Whoosh==2.5.6
django-mptt==0.6.1
lepl==5.1.3
rfc6266==0.0.4
html5lib==0.999
bleach==1.4
email-reply-parser==0.3.0
django-extensions==1.4.0  # used for debugging and UUIDField
shortuuid==0.4.2  # for ShortUUIDField in django-extensions
requests==2.4.1
hellosign-python-sdk==3.2
oauth2client==1.4.4
django-ipware==0.1.0

# Sphinx
Sphinx==1.3b1
Jinja2==2.7.3
Pygments==1.6
docutils==0.12
snowballstemmer==1.2.0
Babel==1.3
MarkupSafe==0.23

### Debugging
django-concurrent-server==0.1.4
django-debug-toolbar==1.2.1
sqlparse==0.1.11  # django-debug-toolbar requirement

### Styleguide
pykss==0.4
stripe==1.19.0
```

**Comment by brettcannon** [comment]
Distlib 0.2.0 has a bug where it doesn't like projects that only have preleases and yet you specify you don't want to carry about them (as caniusepython3 does). I have filed a bug for it at https://bitbucket.org/pypa/distlib/issue/59/ . If you comment out recaptcha==1.0rc1 and django-recaptcha-field==1.0b2 then you get a successful run:
```
You need 17 projects to transition to Python 3.
Of those 17 projects, 15 have no direct dependencies blocking their transition:

  django-celery-transactions
  django-concurrent-server
  django-discover-jenkins
  django-include-strip-tag
  django-ipware
  django-storages
  docx
  hellosign-python-sdk
  oauth2
  pdfminer
  pykss
  reportlab
  lepl (which is blocking rfc6266)
  m2crypto (which is blocking tender_multipass)
  xlsxcessive
```

**Comment by brettcannon** [comment]
I'm going to leave this issue open but I consider it solved since this is a distlib bug that only manifests itself if you specify a project who has never released a full version before. If this becomes an issue I will catch the AttributeError but I would rather not have to take such a drastic move.

**Comment by StoicLoofah** [comment]
Great, thanks so much for investigating here! I can easily work around it with this in mind, so I'm fine if you want to close it.

**Comment by brettcannon** [comment]
Other people have run into this, so it was fixed by @rawrgulmuffins in 81fe6b61057e43d16c3eba9853a5affa57eb5cfc
